projects/batfish/src/test/java/org/batfish/bdp/AbstractRibTest.java

```diff
@@ -0,0 +1,154 @@
+package org.batfish.bdp;
+
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.emptyIterableOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import org.batfish.datamodel.Ip;
+import org.batfish.datamodel.Prefix;
+import org.batfish.datamodel.StaticRoute;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test the AbstractRib tree logic. To avoid worrying about route preference comparisons, testing
```

Comment:
Simpler:

```java
/** Tests of {@link AbstractRib}. */
```

Then move the bit about StaticRoute to the `_rib` variable.

My general rule here is that public Javadoc is for people looking at "What does X do?" whereas private documentation is for people reading the code ("how does X work?").

[And, I don't think I care how we test AbstractRib until I'm reading the code.]